webapp/src/components/EspOta.vue

```diff
@@ -0,0 +1,139 @@
+            </div>
+            </v-col>
+        </v-row>
+        <v-row justify="center" dense>
+            <v-col cols="10">
+            <div class="ma-auto">Current Version: {{ oldFwVersion }}</div>
+            </v-col>
+        </v-row>
+        <v-row justify="center">
+            <v-col cols="10">
+            <v-file-input
+                v-model="file"
+                label="File input"
+                outlined
+                dense
+                :show-size="1000"
+            ></v-file-input>
+            </v-col>
+        </v-row>
+        <v-row justify="center" dense>
+            <v-col cols="2">
+            <v-btn
+                :loading="uploading"
+                @click="start()"
+            >Start</v-btn>
+            </v-col>
+        </v-row>
+        <v-row justify="center">
+            <v-alert
+            v-model="alert"
+            dense
+            outlined
+            type="error"
+            transition="scale-transition"
+            dismissible
+            >{{ status }}</v-alert
+            >
+            <v-alert
+            v-model="success"
+            dense
+            outlined
+            type="success"
+            transition="scale-transition"
+            dismissible
+            >{{ status }}</v-alert
+            >
+        </v-row>
+    </v-card-text>
+  </v-card>
+</template>
+
+<script>
+export default {
+  props: ['deviceID'],
+  data() {
+    return {
+      uploading: false,
+      file: null,
+      status: null,
+      alert: false,
+      success: false,
+      oldFwVersion: null,
+      NewFwVersion: null,
+    };
+  },
+  mounted() {
+    this.fetchData("old");
+  },
+  methods: {
+    start: function () {
+      this.uploading = !this.uploading;
+      var reader = new FileReader();
+      if (!this.file) {
+        this.status = "No file selected";
+        this.showAlert();
+        this.uploading = !this.uploading;
+        return;
+      }
+      reader.readAsArrayBuffer(this.file);
+      reader.onload = () => {
+        this.$ajax
+          .post("api/v1/ota/" + this.deviceID, reader.result)
+          .then((res) => {
+            this.fetchData("new");
+          })
+          .catch((error) => {
+            this.status =
+              error.response.data?.error + " Statuscode: " + error.response.status;
+            this.uploading = !this.uploading;
+            this.showAlert();
+          });
+      };
+    },
+    fetchData: function (target) {
+      this.$ajax
+        .get("api/v1/ts/" + this.deviceID + "/info")
+        .then((res) => {
+          if (target == "old") {
+            this.oldFwVersion = res.data.FirmwareVersion;
+          } else if (target == "new") {
+            this.NewFwVersion = res.data.FirmwareVersion;
+            this.status =
+              "Updated successfully from " +
+              this.oldFwVersion +
+              " --> " +
+              this.NewFwVersion;
```

Comment:
This does not seem to work, it keeps the old version:

![image](https://user-images.githubusercontent.com/17674105/128870685-51192eed-bc61-4999-b981-87a6d54b246c.png)

Or is there some special treatment of the firmware image necessary to make it work?

Reply:
If you are building the same commit, it will not change

Reply:
I created a dummy commit for the new firmware image and it was properly recognized after refreshing the page.

![image](https://user-images.githubusercontent.com/17674105/128889419-6295bd3f-ac32-454f-a294-f6d22e85fb41.png)

Reply:
I think the problem is that the request is fired before the esp actually rebooted. I added a 5s delay to wait for reboot.

Reply:
It works now, really cool! Going to merge :)